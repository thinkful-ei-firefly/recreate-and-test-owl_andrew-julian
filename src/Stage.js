import React from 'react';
import Participant from './Participant';
import './stage.css';



function Stage(props) {
  
  const isOnStage = props.participants.filter(Participant => Participant.onStage);

  return (
    <div className="Stage">
        {isOnStage.map((stageArr) =>
        
            <Participant
                key={stageArr.id}
                id={stageArr.id}
                name={stageArr.name}
                avatar={stageArr.avatar}
                inSession={stageArr.inSession}
                onStage={stageArr.onStage}
                />
            )}
    </div>
  );
}

export default Stage;