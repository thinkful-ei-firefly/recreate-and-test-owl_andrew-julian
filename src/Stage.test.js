import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <Stage
    participants={[{
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    }]}
        />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Stage
        participants={[{
            id: 1,
            name: 'Koren Templeton',
            avatar:
                'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true
        }]}
            />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });
