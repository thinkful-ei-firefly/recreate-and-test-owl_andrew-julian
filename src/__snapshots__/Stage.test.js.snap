// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Stage"
>
  <div
    className="participant inSession"
    id={1}
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <h4>
      Koren Templeton
    </h4>
    <div
      className="onStageStatus onStage"
    >
      <h6>
        â€¢
      </h6>
      <h5>
        on stage
      </h5>
    </div>
  </div>
</div>
`;
